### ISSUE TITLE
spades-kmercount fails to output result files

### ISSUE THREAD

**Issue by rec3141** [ablab/spades#197]
```
% ~/apps/SPAdes-3.13.0-Linux/bin/spades-kmercount corrected/trimmed_Plate1-11E-22_S85_L001_R_unpaired.00.0_0.cor.fastq.gz
  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 155)   Starting SPAdes k-mer counting engine, built from refs/heads/spades_3.13.0, git revision 8ea46659e9b2aca35444a808db550ac333006f8b
  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 157)   K-mer length set to 21
  0:00:00.000     4M / 4M    INFO    General                 (main.cpp                  : 158)   # of threads to use: 56
  0:00:00.004     4M / 4M    INFO    General                 (kmer_index_builder.hpp    : 117)   Splitting kmer instances into 896 files using 56 threads. This might take a while.
  0:00:00.006     4M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 1024
  0:00:00.006     4M / 4M    INFO    General                 (kmer_splitters.hpp        :  97)   Using cell size of 74898
  0:00:04.613    30G / 30G   INFO   K-mer Splitting          (main.cpp                  :  94)   Processing corrected/trimmed_Plate1-11E-22_S85_L001_R_unpaired.00.0_0.cor.fastq.gz
  0:00:39.712    31G / 31G   INFO   K-mer Splitting          (main.cpp                  : 103)   Processed 10690 reads
  0:00:39.713    31G / 31G   INFO   K-mer Splitting          (main.cpp                  : 108)   Total 10690 reads processed
  0:00:41.215   228M / 31G   INFO    General                 (kmer_index_builder.hpp    : 120)   Starting k-mer counting.
  0:00:42.609   228M / 31G   INFO    General                 (kmer_index_builder.hpp    : 127)   K-mer counting done. There are 2484146 kmers in total.
  0:00:42.609   228M / 31G   INFO    General                 (kmer_index_builder.hpp    : 133)   Merging temporary buckets.
  0:00:45.325   228M / 31G   INFO    General                 (kmer_index_builder.hpp    : 150)   Merging final buckets.
  0:00:45.489   228M / 31G   INFO    General                 (main.cpp                  : 174)   K-mer counting done, kmers saved to ./kmer_counter_zXrhqJ/final_kmers_9HG368

% ls -l ./kmer_counter_zXrhqJ/final_kmers_9HG368
ls: cannot access ./kmer_counter_zXrhqJ/final_kmers_9HG368: No such file or directory
```

**Comment by asl** [comment]
Yeah, it seems we have a bug in the kmercount example that does not the output file in the proper place and therefore is deleted.

What is your plan with kmercount?

**Comment by rec3141** [comment]
I'm just testing various kmer counters to use in a metagenomic binning pipeline.

**Comment by asl** [comment]
Ah, ok. Well, currently it's merely the example of using SPAdes internals to perform k-mer counting and has some limitation.  Though it could be extended / specialized to a particular use.

**Comment by lyy005** [comment]
I'm using kmer count for estimating genome size and I have the same issue. No output file can be found. Thanks!

**Comment by asl** [comment]
Yes, apparently it emits the result to temporary file. We will address this issue in the next SPAdes release.

**Comment by asl** [comment]
The issue is certainly fixed in SPAdes 3.15.2
